src/main/java/ml/shifu/shifu/core/dtrain/dt/DTMaster.java

```diff
@@ -896,6 +907,14 @@ public void init(MasterContext<DTMasterParams, DTWorkerParams> context) {
 
         this.toDoQueue = new LinkedList<TreeNode>();
 
+        if(validParams.containsKey("EnableEarlyStop")){
+            this.earlyStopEnabled = Boolean.valueOf(validParams.get("EnableEarlyStop").toString());
+        }
+
+        if(this.earlyStopEnabled){
+            this.dtEarlyStopDecider = new DTEarlyStopDecider(this.maxDepth);
```

Comment:
These two 'if condition' better not combined for two reasons. 
First the second 'if condition' can not be saved. Because the first 'if condition' is used to check if the params contain key "EnableEarlyStop", it does not gurantee the value is true or false. So even if we put the second 'if condition' into the first one, 'if condition' is still needed.
Second, issue may caused in some condition. If we put the second 'if condition' into the first one, then even if the default value of earlyStopEnabled set to be true in the top of DTMaster, this decider would not be initialized.